docs/docs: update upgrading to mention redis storage backend (#1172)
pomerium/pomerium: docs/docs/upgrading.md

### Cache Service

The cache service now stores user and session data in storage backend (rather than in a browser cookie). There are two storage
backend types supported, `memory` or `redis`.

> As a consequence running more than one instance of the cache service is no longer supported.

Reviewer: ```suggestion
Running more than one instance of the `memory` type cache service is not supported.
```

For `memory` storage, restarting the cache service will result in all users having to re-login. The in-memory database used by the cache
service can be found here: [internal/databroker/memory](https://github.com/pomerium/pomerium/tree/master/internal/databroker/memory).

The `redis` storage can be used for persistent data, you can see all [storage backend configuration here].

> You can see the new [databroker gRPC interface](https://github.com/pomerium/pomerium/blob/master/pkg/grpc/databroker/databroker.proto)

Reviewer: Might be worth also referencing the storage interface to invite contributions at that level.
